cleaner.py

```diff
@@ -0,0 +1,194 @@
+        $ sudo -H -u certs bash
+        (certs) certs@learning-tribes:~$ source /edx/app/certs/venvs/certs/bin/activate
+        (certs) certs@learning-tribes:~$ /edx/app/certs/venvs/certs/bin/python /edx/app/certs/certificates/cleaner.py --target_type=[s3|dfs] --dryrun=[true|false]
+        [INFO] Dryrun Mode=True | cleaning AWS/S3 files, Bucket Name=lt-learning-customer2-default
+        Done !
+
+"""
+from abc import ABCMeta, abstractmethod
+from argparse import ArgumentParser
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+import os
+import shutil
+from sys import exit as process_terminate
+from traceback import format_exc
+import uuid
+
+import boto.s3
+from boto.s3.key import Key
+
+import settings
+
+
+class _CleanerInterface(object):
+    """Regulate all subclass of cleaner
+    """
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def run(self):
+        """Execute cleaning task
+        """
+        raise NotImplementedError
+
+
+class DFSCleaner(_CleanerInterface):
+    """Clean unwanted files in DFS ( folders: /downloads + /cert ).
+        And we can specify the `End Date` of date range in the process.
+    """
+    TARGET_ROOT_FOLDERS = [                 # Target folders where we want to clean
+        '/edx/var/certs/www-data/downloads',
+        '/edx/var/certs/www-data/cert'
+    ]
+    PERIOD_START_DATE = datetime(2020, 2, 15)
+    PERIOD_END_DATE = None
+
+    def __init__(self, dryrun=True):
+        self.PERIOD_END_DATE = datetime.now() - relativedelta(
+            months=int(input('Please enter Month number of files which you wanna to remain: '))
+        )
+        print('Please enter start date : ')
+        self.PERIOD_START_DATE = datetime(int(input('Year:')), int(input('Month:')), int(input('Day:')))
+        self.dryrun = dryrun
+        print(
+            '[INFO] Dryrun Mode={} | cleaning "DFS" files from {} to {} in folders : {}'.format(
+                dryrun,
+                self.PERIOD_START_DATE, self.PERIOD_END_DATE,
+                ' + '.join(self.TARGET_ROOT_FOLDERS)
+            )
+        )
+
+    def delete_resources_in_range(self, root_folder):
+        for folder_name in os.listdir(root_folder):
+            resource_folder = os.path.join(root_folder, folder_name)
+            if os.path.isdir(resource_folder):
+                _mod_time = datetime.fromtimestamp(os.path.getmtime(resource_folder))
+                if self.PERIOD_START_DATE < _mod_time < self.PERIOD_END_DATE:
+                    print('[INFO] DELETING {} (Modified: {})'.format(resource_folder, _mod_time))
+                    if self.dryrun == False:
+                        shutil.rmtree(resource_folder)  # Delete folder and its contents
+
+    def run(self):
+        for _root_folder in self.TARGET_ROOT_FOLDERS:
+            print('[INFO] ################# Root folder {} #################'.format(_root_folder))
+            self.delete_resources_in_range(_root_folder)
```

Comment:
Even though there's a bit more going on here than in `S3LearnerCertPNGCleaner`, we don't really need to create a separate function.